File: PythonAnswers/ClimbingStairs.py
# Leetcode problem 70
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        self.my_map = {1:1, 2:2, 3:3}
        return self.recursiveSolution(n)
    
    def recursiveSolution(self, n):
        if n not in self.my_map:
            self.my_map[n] = self.recursiveSolution(n - 1) + self.recursiveSolution(n - 2)

        return self.my_map[n]
    
if __name__ == "__main__":
    solution = Solution()
    
    for i in range(1, 100):
        print("test #" + str(i) + ": " + str(solution.climbStairs(i)))

File: JavaAnswers/DynamicProgramming/ClimbingStairs.java
import java.util.HashMap;

// LeetCode problem 70
public class ClimbingStairs {
    HashMap<Integer, Integer> map = new HashMap<>();

    public static void main(String[] args) {
        ClimbingStairs climbingStairs = new ClimbingStairs();

        for (int i = 1; i < 46; i++) {
            System.out.println("test #" + i + ": " + climbingStairs.climbStairs(i));
        }
    }

    public int climbStairs(int n) {
        map.put(1, 1);
        map.put(2, 2);
        map.put(3, 3);
        return recursiveSolution(n);
    }

    public int recursiveSolution(int n) {
        if (map.get(n) == null) {
            map.put(n, recursiveSolution(n - 1) + recursiveSolution(n - 2));
        }

        return map.get(n);
    }
}
